packages/slice-machine/src/legacy/lib/builders/CustomTypeBuilder/UIDEditor/index.tsx

```diff
@@ -0,0 +1,144 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActionButton,
+  DialogActions,
+  DialogCancelButton,
+  DialogContent,
+  DialogHeader,
+  FormInput,
+  Icon,
+  Text,
+} from "@prismicio/editor-ui";
+import {
+  Group,
+  NestableWidget,
+  UID,
+} from "@prismicio/types-internal/lib/customtypes";
+import { Formik } from "formik";
+import { type FC, useState } from "react";
+import type { AnyObjectSchema } from "yup";
+
+import { updateField } from "@/domain/customType";
+import { useCustomTypeState } from "@/features/customTypes/customTypesBuilder/CustomTypeProvider";
+import {
+  CustomTypes,
+  type TabField,
+  TabFieldsModel,
+} from "@/legacy/lib/models/common/CustomType";
+import { Widgets } from "@/legacy/lib/models/common/widgets";
+import { type Widget } from "@/legacy/lib/models/common/widgets/Widget";
+
+interface UIDEditorProps {
+  field: { key: string; value: TabField };
+}
+
+export const UIDEditor: FC<UIDEditorProps> = ({ field }) => {
+  const [isOpen, setOpen] = useState(false);
+  const [isSaving, setSaving] = useState(false);
+
+  const { customType, setCustomType } = useCustomTypeState();
+  const customTypeSM = CustomTypes.toSM(customType);
+
+  const sectionId =
+    customTypeSM.tabs.find((tab) => {
+      return tab.value.find((f) => f.key === field.key);
+    })?.key ?? "";
+
+  const saveUIDLabel = (label: string) => {
+    const widget: Widget<TabField, AnyObjectSchema> = Widgets["UID"];
+    const newValue = widget.create(label);
+
+    const newField: NestableWidget | UID | Group =
+      TabFieldsModel.fromSM(newValue);
+
+    const newCustomType = updateField({
+      customType,
+      previousFieldId: field.key,
+      newFieldId: field.key,
+      newField,
+      sectionId,
+    });
+
+    setCustomType(newCustomType);
+  };
+
+  return (
+    <Dialog
+      open={isOpen}
+      onOpenChange={setOpen}
+      size={{ width: 448, height: 248 }}
+      trigger={
+        <Button
+          color="grey"
+          textColor="placeholder"
+          startIcon="language"
+          sx={{ marginInline: "auto" }}
+        >
+          :uid
```

Comment:
❓ do you know why we use this `:uid` label? I mean lowercase and with the colon.

Reply:
that's how it is in the design, I didn't really think about why 😅

Reply:
haha, sounds good, let me know if you find out!

Reply:
anyways, now it will be dynamic, displaying the actual set value of the label